File: vripper-server/src/main/java/tn/mnlr/vripper/services/HtmlProcessorService.java
package tn.mnlr.vripper.services;

import org.htmlcleaner.CleanerProperties;
import org.htmlcleaner.DomSerializer;
import org.htmlcleaner.HtmlCleaner;
import org.htmlcleaner.TagNode;
import org.springframework.stereotype.Service;
import org.w3c.dom.Document;
import tn.mnlr.vripper.exception.HtmlProcessorException;

@Service
public class HtmlProcessorService {

  public Document clean(String htmlContent) throws HtmlProcessorException {
    try {
      TagNode clean = new HtmlCleaner().clean(htmlContent);
      return new DomSerializer(new CleanerProperties()).createDOM(clean);
    } catch (Exception e) {
      throw new HtmlProcessorException(e);
    }
  }
}

File: vripper-core/src/main/kotlin/me/mnlr/vripper/services/HtmlProcessorService.kt
package me.mnlr.vripper.services

import me.mnlr.vripper.exception.HtmlProcessorException
import org.htmlcleaner.CleanerProperties
import org.htmlcleaner.DomSerializer
import org.htmlcleaner.HtmlCleaner
import org.w3c.dom.Document
import java.io.InputStream

object HtmlProcessorService {
    @Throws(HtmlProcessorException::class)
    fun clean(htmlContent: InputStream): Document {
        return try {
            htmlContent.use {
                val clean = HtmlCleaner().clean(htmlContent)
                DomSerializer(CleanerProperties()).createDOM(clean)
            }
        } catch (e: Exception) {
            throw HtmlProcessorException(e)
        }
    }
}